flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ObjectPath.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import java.util.Objects;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * A database name and object (table/view/function) name combo in a catalog.
+ */
+public class ObjectPath {
+	private final String databaseName;
+	private final String objectName;
+
+	public ObjectPath(String databaseName, String objectName) {
+		checkNotNull(databaseName, "databaseName cannot be null");
+		checkNotNull(objectName, "objectName cannot be null");
```

Comment:
What are the format restrictions for dataBaseName and objectName, such as special characters? e.g.: '.', duo to we using fullName.split("\\."). right?  So, it is better to add some check logic.
I had left the comment in pre-review.

Reply:
I added basic checks for databaseName and objectName (not-null, not-empty). However, I think the comprehensive checks should be left to parser. If API users provide ill-formed names, it's likely they will get errors in other APIs where this ObjectPath instance is used.